Compiler crash: unknown class property works until actually used

<!--
 Thanks for submitting an issue to AssemblyScript! Please take a moment to
 read the contributing guidelines linked below to get off to a good start 🙂
-->
```js


class Hello {
    prop: string
    // world: Inexistent
    world: World
}


// @JSON
// class World {
//     prop2: string
//     prop3: i32
//     cart: Bump
// }

const testFunc = (hello: Hello): string => {
    return 'aa'
}
console.log('hello')

```

gives
```
$ npx asc assembly/test.ts --target release
Whoops, the AssemblyScript compiler has crashed during compile :-(
▌ 
▌ Here is the stack trace hinting at the problem, perhaps it's useful?
▌ 
▌ /my-proj/node_modules/assemblyscript/std/portable/index.js:198
▌     throw new AssertionError(message);
▌           ^
▌ 
▌ AssertionError: assertion failed

```

Issue 1: crash
Issue 2: removing `testFunc` then compiling gives no error. Should error with 'Unknown type World`

--- a/src/resolver.ts
+++ b/src/resolver.ts
@@ -3351,7 +3351,7 @@ export class Resolver extends DiagnosticEmitter {
               let boundInstance = this.resolveProperty(boundPrototype, reportMode);
               if (boundInstance) {
                 let fieldType = boundInstance.type;
-                assert(isPowerOf2(fieldType.byteSize));
+                if (fieldType == Type.void) break; // failed to resolve earlier
                 let needsLayout = true;
                 if (base) {
                   let existingMember = base.getMember(boundPrototype.name);
@@ -3374,10 +3374,12 @@ export class Resolver extends DiagnosticEmitter {
                   }
                 }
                 if (needsLayout) {
-                  let mask = fieldType.byteSize - 1;
+                  let byteSize = fieldType.byteSize;
+                  assert(isPowerOf2(byteSize));
+                  let mask = byteSize - 1;
                   if (memoryOffset & mask) memoryOffset = (memoryOffset | mask) + 1;
                   boundInstance.memoryOffset = memoryOffset;
-                  memoryOffset += fieldType.byteSize;
+                  memoryOffset += byteSize;
                 }
                 boundPrototype.instance = boundInstance;
                 instance.add(boundPrototype.name, boundPrototype); // reports

